src/mbgl/style/style_impl.cpp

```diff
@@ -341,9 +341,24 @@ void Style::Impl::onSourceDescriptionChanged(Source& source) {
 
 void Style::Impl::onSpriteLoaded(std::vector<Immutable<style::Image::Impl>> images_) {
     auto newImages = makeMutable<ImageImpls>(*images);
+    const auto cmp = [](const auto& a, const auto& b) { return a->id < b->id; };
```

Comment:
I'm just thinking if this comparator could be added as friend operator (< and ==) for `style::Image::Impl`, so we could save some lines of code here and in `sprite_parser.cpp` as well.

Reply:
done, added operator <